File: FangZhiHu/app/src/main/java/com/tongqinyuan/fangzhihu/HomePresenterImpl.java
package com.tongqinyuan.fangzhihu;


import android.util.Log;

import javax.inject.Inject;

import qj.ccmtjf.com.lib.HttpDataSourceImpl;
import qj.ccmtjf.com.lib.LogUtil;
import qj.ccmtjf.com.lib.api.ApiResponse;
import qj.ccmtjf.com.lib.entity.ListData;

/**
 * 首页数据presenter实现
 * Created by liyanxi on 11/14/17.
 * Copyright (c) 2017 www.zhengshijr.com. All rights reserved.
 */

public class HomePresenterImpl extends BasePresenter<HomeContact.View> implements HomeContact.Presenter {
    private static final String TAG = HomePresenterImpl.class.getSimpleName();
    @Inject
    public HomePresenterImpl(HttpDataSourceImpl source) {
        super(source);
    }

    @Override
    public void getHomeData() {
        Log.e(TAG,"==发起请求");
        getHttpData(mSource.getInfo(1), new OnRequestListener<ListData>() {
            @Override
            public void onSuccess(ListData response) {
                Log.e(LogUtil.TAG,"==成功="+response.toString());
            }

            @Override
            public void onFailure(ApiResponse<String> errorApiResponse) {
                Log.e(LogUtil.TAG,"==失败="+errorApiResponse.toString());
            }
        });
    }
}

File: FangZhiHu/KotlinOKhttp/src/main/java/com/kotlinokhttp/HomePresenterImpl.kt
package com.kotlinokhttp

import android.util.Log
import com.kotlinokhttp.entity.ListData
import com.ytq.main.common.ApiMethodSourceImpl
import com.ytq.main.common.OnRequestListener
import network.entity.ApiResponse
import javax.inject.Inject

/**
 * Created by tongqinyuan on 2018/3/9.
 */
class HomePresenterImpl @Inject constructor(source: ApiMethodSourceImpl) : BasePresenter<HomeContact.View>(source),HomeContact.Presenter {
    private val TAG = HomePresenterImpl::class.java.simpleName

    override fun getHomeData() {
        getHttpData(mSource!!.getInfo(1), object : OnRequestListener<ListData> {
            override fun onSuccess(response: ListData) {
                Log.e(TAG, "==成功=" + response.toString())
            }

            override fun onFailure(errorApiResponse: ApiResponse<String>) {
                Log.e(TAG, "==失败=" + errorApiResponse.toString())
            }
        })
    }
}